### PR TITLE
Usernames

### PR DESCRIPTION
**Pull request by Oransj**

### CODE REVIEW

File: internal/models/users/users.go [hunk 1]
```diff
@@ -147,26 +147,47 @@ func scanUserFromFullRow(row *sql.Row) (*User, error) {
...
+        	LIMIT 1) AS random_username
+			WHERE users.id = $1
+			RETURNING CONCAT(random_username.adjective, ' ', random_username.noun);`,
+		userID,
+	).Scan(&username)
+
+	return username, err
 }
 
 // Assigns a username to a user in the database
```
Comment (z-rayc) [inline internal/models/users/users.go:173]: update docs?

File: internal/web_server/web/handlers/quiz_pages_handlers.go [hunk 2]
```diff
@@ -31,6 +33,9 @@ func (qph *QuizPagesHandler) RegisterQuizHandlers(e *echo.Group) {
 	e.GET("/fullforte", qph.getFinishedQuizzes)
 	e.GET("/arkiv", qph.getArchivedQuizzes)
 	e.GET("/profil", qph.getQuizProfile)
+
+	e.GET("/username", qph.usernamePage)
```
Comment (z-rayc) [inline internal/web_server/web/handlers/quiz_pages_handlers.go:37]: Bruk norsk URL siden disse er for norske  brukere

File: internal/web_server/web/handlers/quiz_pages_handlers.go [hunk 3]
```diff
@@ -110,3 +115,34 @@ func (qph *QuizPagesHandler) getArchivedQuizzes(c echo.Context) error {
...
+
+	user, error := users.GetUserByID(qph.sharedData.DB, utils.GetUserIDFromCtx(c))
+	if error != nil {
+		return error
+	}
+
+	return utils.Render(c, http.StatusOK, quiz_pages.UsernamePage(user))
+}
+
+// Adds phone number and competition opt-in to user
```
Comment (z-rayc) [inline internal/web_server/web/handlers/quiz_pages_handlers.go:129]: "competition opt-in to user": is it not more accurate to say "leaderboards opt-in status for user"

File: internal/web_server/web/views/components/icons/dice.templ [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package icons
+
+import "fmt"
+
+//Dice icon. Parameters are strokeWidth, stroke, and height and width in pixels.
+//Example: Clock(32, "currentColor", 24, 24)
```
Comment (z-rayc) [inline internal/web_server/web/views/components/icons/dice.templ:6]: The example should be for Dice

File: internal/web_server/web/views/pages/quiz_pages/username.templ [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import "github.com/Molnes/Nyhetsjeger/internal/models/users"
+
+templ UsernamePage(u *users.User) {
+	@layout_components.BaseLayout("Nyhetsjeger - Brukernavn") {
+		<div class="h-screen flex justify-center items-center">
+			<form class="w-1/3 h-3/4 bg-purple-400 flex flex-col items-center rounded-lg" action="" method="post">
+				<h1 class="text-3xl my-4 font-bold">Lag bruker</h1>
+				<div class="w-3/4">
+					<div>
+						<label for="tlfIn" class="inline font-bold mr-4">Mobilnummer</label>
```
Comment (z-rayc) [inline internal/web_server/web/views/pages/quiz_pages/username.templ:14]: Use kebab case for IDs. Also consider making them global constants so you can use them in the Handler side. To ensure no discrepancies. See admin_api.go example use for variables from the dashboard_pages package for input names..

File: internal/web_server/web/views/pages/quiz_pages/username.templ [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+						<button class="inline rounded-full border-gray-700 text-gray-700 border-2 w-6 h-6 font-bold mb-1" type="button">i</button>
+					</div>
+					<input id="tlfIn" class="w-full h-7 rounded px-2" type="tel" placeholder="123 45 678" pattern="^(\\d{2} \\d{2} \\d{2} \\d{2}|\\d{3} \\d{2} \\d{3}|\\d{8})$" name="phonenumber" id="mobilnummer" required/>
```
Comment (z-rayc) [inline internal/web_server/web/views/pages/quiz_pages/username.templ:17]: Multiple elements here have two IDs. Only one ID per element.

**Review by z-rayc — CHANGES_REQUESTED** [review]
Please resolve all my comments.

File: internal/web_server/web/handlers/quiz_pages_handlers.go (hunk 3 above)
Comment (Molnes) [reply internal/web_server/web/handlers/quiz_pages_handlers.go:129]: både og, ekke det sånn at dei e nødt til å være med på leaderboardsa for å være med i konkuransen?

File: internal/models/users/users.go (hunk 1 above)
Comment (Oransj) [reply internal/models/users/users.go]: done

File: internal/web_server/web/handlers/quiz_pages_handlers.go (hunk 3 above)
Comment (Oransj) [reply internal/web_server/web/handlers/quiz_pages_handlers.go]: Changed to "Adds phone number and the leaderboards opt-in status to the user"

File: internal/web_server/web/handlers/quiz_pages_handlers.go (hunk 2 above)
Comment (Oransj) [reply internal/web_server/web/handlers/quiz_pages_handlers.go]: Gjort

File: internal/web_server/web/views/components/icons/dice.templ (hunk 4 above)
Comment (Oransj) [reply internal/web_server/web/views/components/icons/dice.templ]: Done

File: internal/web_server/web/views/pages/quiz_pages/username.templ (hunk 5 above)
Comment (Oransj) [reply internal/web_server/web/views/pages/quiz_pages/username.templ]: done

File: internal/web_server/web/views/pages/quiz_pages/username.templ (hunk 6 above)
Comment (Oransj) [reply internal/web_server/web/views/pages/quiz_pages/username.templ]: done

**Review by z-rayc — APPROVED** [review]

**Pull request merged**